client.go

```diff
@@ -163,25 +167,35 @@ func (c Client) findIDInConfigurationsPayload(body io.ReadCloser, name string) (
 	return "", err
 }
 
+// ListNodes returns a list of dataset agent nodes from Flocker Control Service
+func (c *Client) ListNodes() (nodes []NodeState, err error) {
+	resp, err := c.get(c.getURL("state/nodes"))
+	if err != nil {
+		return []NodeState{}, err
+	}
+	defer resp.Body.Close()
+
+	err = json.NewDecoder(resp.Body).Decode(&nodes)
+	if err != nil {
+		return []NodeState{}, err
+	}
+	return nodes, err
+}
+
 // GetPrimaryUUID returns the UUID of the primary Flocker Control Service for
 // the given host.
 func (c Client) GetPrimaryUUID() (uuid string, err error) {
-	resp, err := c.get(c.getURL("state/nodes"))
+	states, err := c.ListNodes()
 	if err != nil {
 		return "", err
 	}
-	defer resp.Body.Close()
 
-	var states []nodeStatePayload
-	if err = json.NewDecoder(resp.Body).Decode(&states); err == nil {
-		for _, s := range states {
-			if s.Host == c.clientIP {
-				return s.UUID, nil
-			}
+	for _, s := range states {
+		if s.Host == c.clientIP {
+			return s.UUID, nil
 		}
-		return "", errStateNotFound
 	}
-	return "", err
+	return "", fmt.Errorf("No node not found with IP '%s'", c.clientIP)
```

Comment:
Typo "No node not found"
And I'd be tempted to include the the full list of nodes (in JSON format) in the error message to make it easier to debug misconfigurations....but I'll leave that up to you.